src/System.Text.Json/tests/Serialization/ReadValueTests.cs

```diff
@@ -493,5 +494,24 @@ public static void ReadPartial()
             obj = JsonSerializer.Deserialize(ref reader, typeof(int));
             Assert.Equal(1, obj);
         }
+
+        [Theory]
+        [InlineData("0,1")]
+        [InlineData("0 1")]
+        [InlineData("0 {}")]
+        [InlineData("0/**/")]
+        [InlineData("0 /**/")]
+        public static void TooMuchJson(string json)
+        {
+            byte[] jsonBytes = Encoding.UTF8.GetBytes(json);
+
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<int>(json));
+            Assert.Throws<JsonException>(() => JsonSerializer.Deserialize<int>(jsonBytes));
+            Assert.Throws<JsonException>(() => JsonSerializer.DeserializeAsync<int>(new MemoryStream(jsonBytes)).Result);
+
+            // Using a reader directly doesn't throw.
+            Utf8JsonReader reader = new Utf8JsonReader(jsonBytes);
+            JsonSerializer.Deserialize<int>(ref reader);
```

Comment:
Assert `reader.BytesConsumed != jsonBytes.Length` but also  `reader.BytesConsumed != 0` ?

Reply:
Add the assert that the reader at least read something (i.e. `reader.BytesConsumed != 0`).